apps/testrunner/actions/process_results.py

```diff
@@ -11,18 +13,100 @@
 
 class ProcessResponses(Action):
     REQUIRED_PARAMS = (
-        'responses',
+        'results',
     )
 
-    def run(self):
-        logger.info('Starting processing {0} results'.format(len(self.params['responses'])))
-
-        metrics = [float(response.headers['X-Backend-Response-Time']) for response in self.params['responses']]
-        self.result['response_time'] = {
-            'mean': numpy.mean(metrics),
-            'median': numpy.median(metrics),
-            'lowest': numpy.min(metrics),
-            'highest': numpy.max(metrics),
+    __PROFILER_REGEXP = re.compile(r'^\s*([\d\.]+\%)\s+([\d\.]+)\s+(\d+)\s+\-\s+([^\s].*[^\s])\s*$')
+    __MEMCACHE_REGEXP = re.compile(r'MWMemcached::get.*!(HIT|MISS|DUPE)')
+
+    def _parse_metrics(self, raw_data):
+        metrics = {
+            'server_time': 0.0,
+            'query_time': 0.0,
+            'query_master': 0,
+            'query_slave': 0,
+            'memc_time': 0.0,
+            'memc_misses': 0,
+            'memc_hits': 0,
+            'memc_dupes': 0
+        }
+
+        for line in raw_data.splitlines():
```

Comment:
IMHO 33 lines are not so long. Splitting those into separate function will cause it to be too complex and half of those function will be one line long. We can come back to this when we have more metrics and more generic way to gather them.